tests/unit/services/ivaDefectsService.unitTest.ts

```diff
@@ -21,48 +23,218 @@ describe("IVA Defects Service", () => {
     jest.resetModules();
   });
```

Comment:
I'd opt to use Partial<IvaDatabaseService> for creating our mock service. This approach allows us to define only the specific methods of IvaDatabaseService that are actually used by IvaDefectsService in each test and instancing the service in the before each to avoid bleed into other tests
``` typescript
 let target: IvaDefectsService;
  let mockIvaDatabaseService: Partial<IvaDatabaseService>;

  const ivaDatabaseService = {
    getDefectsByEUVehicleCategory: mockGetDefectsByEUVehicleCategory,
    getDefectsByCriteria: mockGetDefectsByCriteria,
  };

  beforeEach(() => {
    jest.clearAllMocks();
    mockIvaDatabaseService = {
      getDefectsByEUVehicleCategory: mockGetDefectsByEUVehicleCategory,
      getDefectsByCriteria: mockGetDefectsByCriteria,
    };
    target = new IvaDefectsService(mockIvaDatabaseService as IvaDatabaseService);
  });
```